src/main/java/io/github/bobocodebreskul/context/scan/utils/ScanUtils.java

```diff
@@ -11,31 +12,53 @@
  * @author Oleh Yakovenko
  */
 public interface ScanUtils {
-    /**
-     * Search all classes, (except enums and records) in given package including nested packages.
-     * @param packagePath - root package for search.
-     * @return Set of found classes
-     */
-    Set<Class<?>> searchAllClasses(String packagePath);
 
-    /**
-     * Search classes which has given annotation or annotations which has given annotation inside.
-     * For example if @Service annotation has @Component annotation, and we're searching
-     * by @Component we also will find all classes with @Service service annotation.
-     * !Note: given annotation must have @Retention(RetentionPolicy.RUNTIME)
-     *
-     * @param packagePathPrefix root package for search.
-     * @param annotation target annotation.
-     * @return Set of all classes with given annotation.
-     */
-    Set<Class<?>> searchClassesByAnnotationRecursively(String packagePathPrefix, Class<? extends Annotation> annotation);
+  /**
+   * Search all classes, (except enums and records) in given package including nested packages.
+   *
+   * @param packagePath - root package for search.
+   * @return Set of found classes
+   */
+  Set<Class<?>> searchAllClasses(String packagePath);
 
-    /**
-     * Search classes by filter in given package including nested packages.
-     * @param packagePathPrefix root package for search.
-     * @param filter filter predicate which applies on found classes
-     * @return Set of all classes which pass the filter.
-     */
-    Set<Class<?>> searchClassesByFilter(String packagePathPrefix, Predicate<Class<?>> filter);
+  /**
+   * Search classes which has given annotation or annotations which has given annotation inside. For
+   * example if @Service annotation has @Component annotation, and we're searching by @Component we
+   * also will find all classes with @Service service annotation. !Note: given annotation must have
+   * @Retention(RetentionPolicy.RUNTIME)
```

Comment:
Seems this line fails one of the github actions maven builds
<img width="867" alt="image" src="https://github.com/bobocode-breskul/bring/assets/17567540/2603d0a5-821f-409f-ab73-a6adbfdedb8d">